packages/base-controller/src/ControllerMessenger.ts

```diff
@@ -533,7 +552,7 @@ export class ControllerMessenger<
       this.events.set(eventType, subscribers);
     }
 
-    subscribers.set(handler, selector);
+    subscribers.set(handler, selector as SelectorFunction<unknown[], V>);
```

Comment:
Why does this need to be cast?

Reply:
We need better typing for `EventSubscriptionMap`. Working on this.

Reply:
Type casting removed and `EventSubscriptionMap` typing improved here: 7baebd7814ea5f65962cb4316825c50d2488bfd6